### ISSUE TITLE
Unholy Claw of Osyluth does not heal while Frozen Heart of Az'heran is disabling healing

### ISSUE THREAD

**Issue by DeckardCainRPC** [ChalkyBrush/roshpit-bug-tracker#1018]
is this because:

1. 0.65% of 100 health is less than 1 HP, and the healing value doesn't round up to the nearest whole number?
2. dad gay?

**Comment by lolwew** [comment]
1HP per hit? Won't it make him nearly unkillable? (arcana chest muti-target)

**Comment by DeckardCainRPC** [comment]
not even close, both against single targets AND multiple targets

instances of physical/magical/pure damage remove 2/5/8 points of health from Frozen Heart's HP pool. E4 gives chernobog 100% evasion, but it costs HP every second, and this drain DOES occur while wearing the trinket. chernobog is already natively "invincible" against enemies with no way to pierce evasion and no spells, and Frozen Heart does not change this; instead, it makes him even more vulnerable to his existing "bad matchups," because 5 or 8 damage from each instance is more than what his 3% steadfast would ordinarily reduce it to.

so against the targets which threaten him most, he's taking more effective HP damage per instance received. (as an additional note, Lightning Enchanted paragons will immediately disintegrate him.) IF each such target attacking him deals damage only once per second - which we know they don't, they attack really fast or have poison DoTs that tick every 0.03 seconds - then chernobog would have to attack 5 or 8 times per second just to avoid *losing* health during that second. the hero's default Base Attack Time of 1.7 yields a maximum attack interval of ~0.21, just shy of 5 attacks per second, meaning Osyluth still wouldn't counteract the damage from a single instance of magic damage per second from just 1 enemy, and it for damn sure would not do anything against pure damage in the same interval.

even with a supplementary bladeslinger boots + wind deity crown with non-arcana E2 and the Glyph which quickens E2 strikes to once every 0.3 seconds, chernobog still only attacks ~13 times per second, which would regenerate HP only slightly faster than Frozen Heart's normal regeneration rate of 10 HP per second. i recognize that fighting multiple targets at once would yield more regeneration, however you must keep in mind that if these targets are all threats to chernobog (again meaning they have spells or retaliatory magic/pure damage passives or pure strike or literally anything besides just autoattacks), then they will ALL be dealing damage to him faster than he can regenerate. in fact, against targets with retaliatory passives, his 13 attacks will either kill him immediately (e.g. pure damage from the Blademail Bears in sea fort) or kill him in just over 1 second of exposure (e.g. magic damage some monsters in the elite Fire Temple wing). worse, each unit he kills is one less unit he's getting Osyluth heals from, and all of this is going on while E drains health every second, independent of any damage he's taking.

**Comment by redledcat** [comment]
Well.
I think there are two correct ways here.
1) Frozen Heart has higher priority than Osyluth.
HP removal from W and E shouldn't affect your HP under Heart effect.
Osyluth won't replenish your HP under Heart effect.
Osyluth tooltip changed to "despite of @Ability2 restriction"
2) Osyluth has highest priority of healing.
HP removal from W and E both has no effect under Heart.
Osyluth can heal 1 HP per hit under Heart effect.

**Comment by Nintai** [comment]
Current State:
The Weapon heals through W, Centuar Helm
The Weapon doesnt heal with Frozen Heart

Thats why 1) -> Tooltip would be wrong

Also im against W and E not removing HP when using Frozen Heart, this would make it pretty strong
But i agree, u could make it so that it heals through Frozen Heart, but maybe change Life to 10000 and dmg to 200, 500, 800 so that weapon can heal 65 hp? 😛

**Comment by redledcat** [comment]
->remove healing through centaur helm
->fix tooltip

**Comment by DeckardCainRPC** [comment]
no, given the choice between that and your earlier Option 2, then i'll take Option 2

i agree with nintai when he says that making W and E not remove HP when using Frozen Heart is probably a little unbalanced, even though i believe Mordiggus gloves also spend HP from Frozen Heart's pool, so it's not like this philosophical question is COMPLETELY unique to Chernobog. but, i will say that healing 1 HP at a time, when the least amount of damage he can take per instance is 2, is not as severely broken as you might think.

what i would like to recommend, is making the change on RPC Test, and seeing if it's broken with my Wind Deity Crown + Bladeslinger Boots cherno i made. that is the maximum possible abuse. if the change makes him too much cheese, then i will definitely agree it is a bad idea, and then i would also support changing one of the tooltips (i can't decide which one) to make it clear that Frozen Heart prevents ALL healing, even if the healing source SAYS it should normally work even while healing is disabled.
